### PR TITLE
Refs #4151 - Use new report API introduced in Foreman 1.11

### PR DESCRIPTION
**Pull request by ares**

### CODE REVIEW

**Comment by domcleal** [comment]
:+1: generally, but I expect the spec will need updating too.

**Comment by ares** [comment]
@domcleal sorry it took me a while to get back to it, should be green now, not sure what "invalid option: --no-80chars-check" in tests mean, some deprecated linter option?

**Comment by domcleal** [comment]
Yes, it's being fixed on the modulesync branch. Tests will continue to fail for the time being.

**Comment by mmoll** [comment]
@ares what's the status here? also, this needs a rebase to fix the tests...

**Comment by ares** [comment]
@mmoll it was waiting for the fix mentioned above, rebased :-)

**Comment by ekohl** [comment]
:+1: on green tests

**Comment by mmoll** [comment]
This doesn't seem to work in my nightly install, is there anything that needs to be done additionally?
```
2016-09-05T21:34:38 b4764cca [app] [I] Started POST "/api/config_reports" for 192.168.200.17 at 2016-09-05 21:34:38 +0200
2016-09-05T21:34:38 b4764cca [app] [I] Processing by Api::V2::ConfigReportsController#create as JSON
2016-09-05T21:34:38 b4764cca [app] [I]   Parameters: {"report"=>"[FILTERED]", "apiv"=>"v2", "config_report"=>{}}
2016-09-05T21:34:38 b4764cca [app] [I] processing report for 
2016-09-05T21:34:38 b4764cca [app] [W] Action failed
 | NoMethodError: undefined method `downcase' for nil:NilClass
 | /usr/share/foreman/app/services/report_importer.rb:55:in `host'
 | /usr/share/foreman/app/services/report_importer.rb:115:in `create_report_and_logs'
 | /usr/share/foreman/app/services/config_report_importer.rb:13:in `create_report_and_logs'
 | /usr/share/foreman/app/services/report_importer.rb:39:in `import'
 | /usr/share/foreman/app/services/report_importer.rb:20:in `import'
 | /usr/share/foreman/app/models/config_report.rb:27:in `import'
 | /usr/share/foreman/app/controllers/api/v2/config_reports_controller.rb:40:in `create'
[...]
```

File: files/foreman-report_v2.rb [hunk 1]
```diff
@@ -58,13 +58,13 @@ def process
           http.key  = OpenSSL::PKey::RSA.new(File.read(SETTINGS[:ssl_key]), nil)
         end
       end
-      req = Net::HTTP::Post.new("#{uri.path}/api/reports")
+      req = Net::HTTP::Post.new("#{uri.path}/api/config_reports")
       req.add_field('Accept', 'application/json,version=2' )
       req.content_type = 'application/json'
       req.body         = {'report' => generate_report}.to_json
```
Comment (domcleal) [inline files/foreman-report_v2.rb]: `'config_report'`

**Comment by ares** [comment]
Thanks, fixed.

**Comment by domcleal** [comment]
Appears to work now:

<pre>
2016-09-06 03:50:16 e0cc63c1 [app] [I] Processing by Api::V2::ConfigReportsController#create as JSON
2016-09-06 03:50:16 e0cc63c1 [app] [I]   Parameters: {"config_report"=>"[FILTERED]", "apiv"=>"v2"}
2016-09-06 03:50:16 e0cc63c1 [app] [I] processing report for foreman.example.com
2016-09-06 03:50:16 e0cc63c1 [app] [I] Imported report for foreman.example.com in 0.15 seconds
2016-09-06 03:50:16 e0cc63c1 [app] [I]   Rendered api/v2/config_reports/create.json.rabl (31.1ms)
2016-09-06 03:50:16 e0cc63c1 [app] [I] Completed 201 Created in 236ms (Views: 35.0ms | ActiveRecord: 105.7ms)
</pre>

**Comment by mmoll** [comment]
works fine, merged, díky @ares!

**Pull request merged**